packages/valory/skills/decision_maker_abci/behaviours/sampling.py

```diff
@@ -231,9 +273,27 @@ def _benchmarking_inc_day(self) -> Tuple[bool, bool]:
 
         return benchmarking_finished, day_increased
 
+    def get_active_bets(self) -> Generator:
+        """Get the active bets."""
+        trades = yield from self.fetch_trades(
+            self.synchronized_data.safe_contract_address.lower(), 0.0, time.time()
+        )
+        if trades is None:
+            return []
+
+        user_positions = yield from self.fetch_user_positions(
+            self.synchronized_data.safe_contract_address.lower()
+        )
+        if user_positions is None:
+            return []
+
+        balances = get_bet_id_to_balance(trades, user_positions)
+        return balances
```

Comment:
I wonder whether we really need to fetch that info, or we could use the local `bets` property and filter for bets with a non-zero `invested_amount`. The balances should match the `invested_amount`.

Though I still find your solution better as it also covers cases in which the file is deleted or whatever.

Reply:
No action needed; feel free to resolve this conversation.